### PR TITLE
make setting `--backend:xxx` also set exception handling strategy

### PR DESCRIPTION
**Pull request by cooldome**
Currently `nim check a.nim` fails if `a` contains importing and handling of native cpp exceptions. `nim check` isn't aware of cpp exceptions. I thought `nim check --backend:cpp` will fix it, but this PR is required to make it work.

### CODE REVIEW

**Comment by Araq** [comment]
For some reason this fails:

tests/stdlib/tosproc.nim

File: compiler/commands.nim [hunk 1]
```diff
@@ -444,7 +466,7 @@ proc processSwitch*(switch, arg: string, pass: TCmdLinePass, info: TLineInfo;
   of "backend", "b":
     let backend = parseEnum(arg.normalize, TBackend.default)
     if backend == TBackend.default: localError(conf, info, "invalid backend: '$1'" % arg)
-    conf.backend = backend
+    handleBackend(conf, backend)
```
Comment (timotheecour) [inline compiler/commands.nim:470]: this prevents doing:
`nim r -b:c -b:cpp main`
this is useful to provide a way to override a backend in an command line, eg:
you have a command line template `nim r -b:c` and then you simply add a flag to override via `nim r -b:c -b:cpp`

right now there's a bug (https://github.com/timotheecour/Nim/issues/175) so this doesn't quite work yet (I can fix it unless you want to fix it), but your patch makes it worse because `handleBackend` has side effect. So we should fix both https://github.com/timotheecour/Nim/issues/175 and `nim check`

TLDR: parsing the flag `--backend` should have zero side effect beyond setting `conf.backend = backend`, so that it can be overridden in subsequent flags


(eg, smthg similar is used for runnableExamples, where we have a template command that can be overriden with `--doccmd:-b:cpp -d:foo`)

**Review by timotheecour — CHANGES_REQUESTED** [review]

File: compiler/commands.nim (hunk 1 above)
Comment (cooldome) [reply compiler/commands.nim:470]: This is definitely wrong:
  parsing the flag --backend should have zero side effect beyond setting conf.backend = 
  backend

Comment (timotheecour) [reply compiler/commands.nim:470]: it's by design so that it can be overridden and so that it plays nice with other commands; the side effects are deferred until the command is processed (eg `nim doc` etc).

I'm fixing this issue + others in https://github.com/nim-lang/Nim/pull/14363 which should supersede your PR

**Comment by timotheecour** [comment]
@cooldome ok to close this? (see https://github.com/nim-lang/Nim/pull/14363)

**Comment by cooldome** [comment]
Great thanks, closing this PR
